dev/com.ibm.ws.kernel.service/src/com/ibm/wsspi/kernel/service/utils/SerializableSchedule.java

```diff
@@ -215,44 +296,86 @@ public boolean equals(Object o) {
+     */
+    public boolean withinSchedule(LocalDateTime time) {
+        return withinSchedule(time.atZone(timezone));
+    }
+
+    /**
+     * <pre>
+     * Method to determine if a ZonedDateTime is within the schedule represented by this object.
      *
-     * Example2: if the schedule is "MON 8:00-17:00" and you are checking the schedule ever 6 minutes then calling
-     * this method between 7:57 and 17:03 will return true Monday.
+     * To check the current server time use {@link #withinSchedule()} instead.
+     * To check a LocalDateTime without the timezone component use {@link #withinSchedule(LocalDateTime)} instead.
      *
-     * NOTE: if the schedule is "MON 17:00-8:00" this is equivalent to "MON 17:00-0:00" and "MON 00:00-8:00"
+     * This method will convert the time provided to the timezone that this schedule uses.
+     * Then it will check if that time is within the schedule.
      *
+     * If this schedule represents the moment the server starts, then withinSchedule will always return false.
      * </pre>
      *
-     * @param marginNS - Amount of nanoseconds to set as a margin
-     * @return true we are within the schedule, false otherwise.
+     * @param time - ZonedDateTime
+     * @return boolean - true the time provided is within the scheduled represented by this object, false otherwise.
      */
-    public boolean isNowWithinTheSchedule(long marginNS) {
+    public boolean withinSchedule(ZonedDateTime time) {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) {
+            Tr.entry(tc, "withinSchedule", time);
+        }
 
-        // If day is not valid don't check time
-        if (validDays.contains(LocalDate.now().getDayOfWeek()))
+        if (isStartup()) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "withinSchedule", "No moment in time will ever be within this schedule since it represents the server startup time.", this);
+            }
             return false;
+        }
 
-        LocalTime startAdjusted = startTime.minusNanos(marginNS / 2);
-        LocalTime endAdjusted = endTime.plusNanos(marginNS / 2);
-        LocalTime now = LocalTime.now();
+        //Adjust provided time to the same timezone as this object
+        if (time.getZone() != timezone) {
+            time = time.withZoneSameInstant(timezone);
+        }
+
+        final boolean result;
+
+        // If day is not valid don't check time
+        if (!validDays.contains(time.getDayOfWeek())) {
+            result = false;
+        } else {
+            LocalTime check = time.toLocalTime();
+            result = startTime.isBefore(endTime) ? // Dont need to worry about wrapped time
+                            (startTime.isBefore(check) && check.isBefore(endTime)) : // [ xxx start --- check --- end xxx ]
+                            (check.isBefore(endTime) || startTime.isBefore(check)); //  [ --- check --- end xxx start --- check --- ]
```

Comment:
It seems like performing the comparisons in LocalTime would not properly consider DaylightSavingTime of the zone.  I would have expected the LocalTime + configured zone to be made into a ZonedDateTime, which could then be compared to the time argument (ZonedDateTime) that is supplied this method.